sensu/plugins/metrics-nova-project-usage.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+
+import sys
+import time
+import argparse
+import traceback
+from datetime import datetime, timedelta
+
+import shade
+import novaclient
+
+OK = 0
+WARNING = 1
+CRITICAL = 2
+UNKNOWN = 3
+
+
+class CloudMetrics(object):
+    def __init__(self):
+        self.cloud = shade.openstack_cloud()
+
+    def get_projects(self):
+        return {
+            project['id']: project['name']
+            for project in self.cloud.search_projects()
+        }
+
+    def get_usage(self, project_ids, minutes, hours, days):
+        stop = datetime.now()
+        start = stop - timedelta(minutes=minutes, hours=hours, days=days)
+        usage = []
+        for id_ in project_ids:
+            _usage = (
+                self.cloud.nova_client.usage.get(id_, start, stop).to_dict())
+            _usage['id'] = id_
+            usage.append(_usage)
+        return usage
+
+    def graphite_print(self, quotas):
+        utime = time.time()
+        metric_path = "usage.nova.project_id.{id}.{name}"
+        outstr = "{metric_path} {value} {time}"
+
+        for quota in quotas:
+            id_ = quota['id']
+            for key, val in quota.items():
+                if key in ['id', 'start', 'stop', 'server_usages']:
+                    continue
+                path = metric_path.format(id=id_, name=key)
+                print(outstr.format(metric_path=path, value=val, time=utime))
+
+    def run(self, args):
+        projects = self.get_projects()
+        try:
+            usage = self.get_usage(
+                projects, args.minutes, args.hours, args.days
+            )
+        except novaclient.exceptions.Forbidden:
```

Comment:
Do we still get a `novaclient` exception here, if we're going through via shade?